internal/api/token_oidc.go

```diff
@@ -179,10 +184,10 @@ func (a *API) IdTokenGrant(ctx context.Context, w http.ResponseWriter, r *http.R
 	}
 
 	if !correctAudience {
-		return oauthError("invalid request", "Unacceptable audience in id_token")
+		return oauthError("invalid request", fmt.Sprintf("Unacceptable audience in id_token: %v", idToken.Audience))
```

Comment:
Had an initial concern around whether the audience is sensitive.

External references suggests it [should be external facing](https://www.ory.sh/docs/hydra/guides/audiences) and suggests [it is okay to show](https://community.auth0.com/t/audience-aud-claim-mismatch-in-the-id-token/37692)